sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -1076,4 +1079,168 @@ private AccessPolicy toAccessPolicy(TableAccessPolicy tableAccessPolicy) {
             .setStart(tableAccessPolicy.getStartsOn())
             .setPermission(tableAccessPolicy.getPermissions());
     }
+
+    /**
+     * Executes all operations within the batch inside a transaction. When the call completes, either all operations in
+     * the batch will succeed, or if a failure occurs, all operations in the batch will be rolled back. Each operation
+     * in a batch must operate on a distinct row key. Attempting to add multiple operations to a batch that share the
+     * same row key will cause an error.
+     *
+     * @param transactionalBatch A list of {@link TableTransactionAction transaction actions} to perform on entities
+     * in a table.
+     *
+     * @return A reactive result containing a list of {@link BatchOperationResponse sub-responses} for each operation
+     * in the batch.
+     *
+     * @throws IllegalStateException If no operations have been added to the batch.
+     * @throws TableServiceException If any operation within the batch fails. See the documentation for other client
+     * methods in {@link TableAsyncClient} to understand the conditions that may cause a given operation to fail.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public synchronized Mono<List<BatchOperationResponse>> submitTransaction(List<TableTransactionAction> transactionalBatch) {
```

Comment:
Should we call this parameter `transactionActions`?

Reply:
We could, especially if we want to remove references to the work `batch`, but .NET would have to do so as well. Honestly I just grabbed the name .NET was using to be consistent :P

Reply:
.NET names it transactionActions

https://github.com/Azure/azure-sdk-for-net/blob/74293800392e75c160d841784e01ae74b1bd93a9/sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs#L45